client/app/visualizations/counter/index.js

```diff
@@ -41,6 +41,11 @@ function CounterRenderer($timeout) {
           const targetRowNumber = getRowNumber($scope.visualization.options.targetRowNumber, queryData.length);
           const counterColName = $scope.visualization.options.counterColName;
           const targetColName = $scope.visualization.options.targetColName;
+          const counterLabel = $scope.visualization.options.counterLabel;
+
+          if (counterLabel) {
+            $scope.counterLabel = counterLabel;
+          }
```

Comment:
Just to clarify, and maybe I misunderstood the feature, does this actually keep falling back to the visualization name as I assumed when the `<counter-name>` element's content is changed to `counterLabel`?

Reply:
Currently master uses the viz name for the Counter label, and I wanted to make sure that this is backward compatible. In my previous comment about the placeholder I assumed that your patch would stay backward-compatible, but I think this needs another change so we don't see suddenly a lot of Counters no have a label anymore upon upgrade.

Reply:
Makes sense to me. Please check the change I made.

Reply:
@ralphilius LGTM